packages/circuit-ui/components/SidePanel/SidePanel.tsx

```diff
@@ -15,147 +15,171 @@
-  isStacked,
-  onBack,
-  onClose,
-  ...props
-}: SidePanelProps) => {
-  if (
-    process.env.NODE_ENV !== 'production' &&
-    process.env.NODE_ENV !== 'test'
-  ) {
-    if (!isSufficientlyLabelled(closeButtonLabel)) {
-      throw new AccessibilityError(
-        'SidePanel',
-        'The `closeButtonLabel` prop is missing or invalid.',
+export const SidePanel = forwardRef<HTMLDialogElement, SidePanelProps>(
+  (props, ref) => {
+    const {
+      open,
+      backButtonLabel,
+      children,
+      closeButtonLabel,
+      headline,
+      isMobile,
+      onBack,
+      onCloseEnd,
+      onClose,
+      preventOutsideClickClose,
+      animationDuration = 0,
+      preventEscapeKeyClose,
+      className,
+      ...rest
+    } = useI18n(props, translations);
+    {
+      const [animationClass, setAnimationClass] = useState<string | undefined>(
+        classes.closed,
       );
-    }
-    if (onBack && !isSufficientlyLabelled(backButtonLabel)) {
-      throw new AccessibilityError(
-        'SidePanel',
-        'The `backButtonLabel` prop is missing or invalid.',
+      const [isHeaderSticky, setHeaderSticky] = useState(false);
+      const headerAriaId = useId();
+
+      const [, setAnimating] = useAnimation();
+
+      const triggerAnimation = useCallback(
+        (isOpening: boolean, callBack?: OnBack | OnClose) => {
+          setAnimating({
+            duration: animationDuration,
+            onStart: () => {
+              if (isOpening) {
+                setAnimationClass(
+                  animationDuration > 0
+                    ? sharedClasses.animationSlideRightIn
+                    : undefined,
+                );
+              } else {
+                void callBack?.();
+                setAnimationClass(sharedClasses.animationSlideRightOut);
+              }
+            },
+            onEnd: () => {
+              if (isOpening) {
+                setAnimationClass(undefined);
+              } else {
+                setAnimationClass(classes.closed);
+                onCloseEnd?.();
+              }
+            },
+          });
+        },
+        [setAnimating, animationDuration, onCloseEnd],
       );
-    }
-  }
 
-  const [isHeaderSticky, setHeaderSticky] = useState(false);
-  const headerAriaId = useId();
+      // biome-ignore lint/correctness/useExhaustiveDependencies: Not sure why this effect is necessary
+      useEffect(() => {
+        setHeaderSticky(false);
+      }, [isMobile]);
 
-  // biome-ignore lint/correctness/useExhaustiveDependencies: Not sure why this effect is necessary
-  useEffect(() => {
-    setHeaderSticky(false);
-  }, [isMobile]);
+      const handleScroll: UIEventHandler<HTMLDivElement> = (event) => {
+        setHeaderSticky(event.currentTarget.scrollTop > 0);
+      };
 
-  const handleScroll: UIEventHandler<HTMLDivElement> = (event) => {
-    setHeaderSticky(event.currentTarget.scrollTop > 0);
-  };
+      const triggerAnimationRef = useLatest(triggerAnimation);
```

Comment:
`triggerAnimation` is wrapped in `useCallback`, so shouldn't change often. Why is the ref wrapper necessary then? One of these performance optimizations is likely redundant.